src/python/pants/backend/python/typecheck/pyright/rules.py

```diff
@@ -71,6 +80,58 @@ class PyrightPartitions(Collection[PyrightPartition]):
     pass
 
 
+@rule_helper
+async def _patch_config_file(
+    config_files: ConfigFiles, venv_dir: str, source_roots: Iterable[str]
+) -> Digest:
+    """Patch the Pyright config file to use the incoming venv directory (from
+    requirements_venv_pex). If there is no config file, create a dummy pyrightconfig.json with the
+    `venv` key populated.
+
+    The incoming venv directory works alongside the `--venv-path` CLI argument.
+
+    Additionally, add source roots to the `extraPaths` key in the config file.
+    """
+
+    if not config_files.snapshot.files:
+        # venv workaround as per: https://github.com/microsoft/pyright/issues/4051
+        return await Get(
+            Digest,
+            CreateDigest(
+                [
+                    FileContent(
+                        "pyrightconfig.json",
+                        f'{{ "venv": "{venv_dir}" }}'.encode(),
+                    )
+                ]
+            ),
+        )
+
+    config_contents = await Get(DigestContents, Digest, config_files.snapshot.digest)
+    new_files: list[FileContent] = []
+    for file in config_contents:
+        if file.path == "pyrightconfig.json":
```

Comment:
Is this supposed to be an `os.path.basename` check, or are you not expecting files at any other location?

Reply:
At the moment, we only support a single pyrightconfig.json (or pyproject.toml) in the root directory. Eventually, we'll need to solve multiple workspaces, but I think that might need to come along with execution environments (which I created a ticket for).

Multiple workspaces/multiple configs will add a bit more craziness, with the whole `venv` patch needing to be applied to each, and possibly each with it's own requirements. 

So as of today, I think just a single config in root makes sense.